gateway/src/node/client_handling/websocket/connection_handler.rs

```diff
@@ -366,6 +365,8 @@ where
     async fn handle_initial_authentication_request(
         &mut self,
         mix_sender: MixMessageSender,
+        mix_receiver: MixMessageReceiver,
```

Comment:
I'm not entirely sure what's the purpose of those arguments. It looks like you're assigning to `final_receiver` the value of `mix_receiver` you just passed in here?

Reply:
This is sort of a hack, because the credential comes as part of the authenticate/register process, immediately after one of those messages. Because of that, two `mpsc` channels are created, instead of one, and only the first one is needed. Introducing a separated state for the credential message should remove this anyway.